packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -317,37 +375,48 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
       'A TextDirection must be provided before a Scrollbar can be painted.',
     );
 
+    final ScrollbarOrientation resolvedOrientation;
+
+    if (scrollbarOrientation == null) {
+      if (_isVertical && textDirection == TextDirection.ltr)
+        resolvedOrientation = ScrollbarOrientation.right;
+      else if (_isVertical && textDirection == TextDirection.rtl)
+        resolvedOrientation = ScrollbarOrientation.left;
+      else
+        resolvedOrientation = ScrollbarOrientation.bottom;
+    }
+    else {
+      resolvedOrientation = scrollbarOrientation!;
+    }
+
     final double x, y;
     final Size thumbSize, trackSize;
     final Offset trackOffset;
 
-    switch (direction) {
-      case AxisDirection.down:
+    assert(_debugAssertIsValidOrientation(resolvedOrientation));
+    switch(resolvedOrientation) {
+      case ScrollbarOrientation.left:
         thumbSize = Size(thickness, thumbExtent);
         trackSize = Size(thickness + 2 * crossAxisMargin, _trackExtent);
-        x = textDirection == TextDirection.rtl
-          ? crossAxisMargin + padding.left
-          : size.width - thickness - crossAxisMargin - padding.right;
+        x = crossAxisMargin + padding.left;
```

Comment:
These are really nice simplifications since you have already taken TextDirection into account. :)